src/main/java/com/google/edith/servlets/LoginServlet.java

```diff
@@ -0,0 +1,70 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.edith.servlets;
+
+import com.google.appengine.api.users.User;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that checks if user is logged in.
+ * if logged in then provides with user information along with logout url.
+ * if not logged in then redirectes to a url to log in.
+ */
+@WebServlet("/login")
+public class LoginServlet extends HttpServlet {
+  
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+
+    if (userService.isUserLoggedIn()) {
+      Gson gson = new Gson();
+      String json = gson.toJson(createUserInfo(userService));
+      response.setContentType("application/json");
+      response.getWriter().println(json);
+    } else {
+      String loginUrl = userService.createLoginURL("/");
+      response.sendRedirect(loginUrl);
+    }
+  }
+  
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.sendRedirect("/index.html");
```

Comment:
If the only thing that this post method is doing is redirecting back to the homepage, do we need it in the class?

Reply:
Yes, it is being called in the future to store user info.